pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1502,6 +1502,27 @@ func (c *K8sOrchestrator) AnnotateVolumeSnapshot(ctx context.Context, volumeSnap
 	return c.updateVolumeSnapshotAnnotations(ctx, volumeSnapshotName, volumeSnapshotNamespace, annotations)
 }
 
+// IsVolumeSnapshotAnnotationPresent checks if the volumesnapshot CR has the given annotation
+// set or not.
+func (c *K8sOrchestrator) IsVolumeSnapshotAnnotationPresent(ctx context.Context, volumeSnapshotName string,
+	volumeSnapshotNamespace string, annotationKey string) (bool, error) {
+	log := logger.GetLogger(ctx)
+	// Retrieve the volumesnapshot CR and verify that it exists
+	volumeSnapshot, err := c.snapshotterClient.SnapshotV1().VolumeSnapshots(volumeSnapshotNamespace).
+		Get(ctx, volumeSnapshotName, metav1.GetOptions{})
+	if err != nil {
+		return false, logger.LogNewErrorf(log, "failed to get the volumesnapshot %s/%s. Error: %+v",
```

Comment:
You might want to handle SnapShot NotFound error. Thoughts ?

Reply:
+1. Good catch.

Reply:
Done. I have updated the code accordingly.